main.cpp

```diff
@@ -81,12 +81,12 @@ struct Temporary
         std::cout << "I'm a Temporary<" << typeid(v).name() << "> object, #"
                   << counter++ << std::endl;
     }
-    ~Temporary(){}
+    ~Temporary() = default;
     //rule of 5 move funcs
-    Temporary(NumericType&& num) { v = std::move(num); }
-    Temporary& operator=(NumericType&& num) 
+    Temporary(Temporary&& num) { v = std::move(num.v); }
```

Comment:
use the member initializer list for constructors, not the in-constructor variable assignment.
```
Temporary(Temporary&& num) : v(std::move(num.v)) { }
```
- [ ] revise